### PR TITLE
[W06H03] Add further tests

### PR DESCRIPTION
**Pull request by julian-hartl**
The new tests cover the following:
- Negative amount of nodes (is possible)
- Out of range calls on isAdj, getAdj and addEdge
- Removal of isAdjFast test

### CODE REVIEW

**Review by TheAleWu — APPROVED** [review]

File: w06h03/test/pgdp/ds/behavior/GraphTest.java [hunk 1]
```diff
@@ -60,31 +50,25 @@ final void directionOfEdgeShouldBeRespected() {
...
     }
 
-    private <T> T switchGraph(T sparse, T dense) {
-        return switch (graphType.getSimpleName()) {
-            case "SparseGraph" -> sparse;
-            case "DenseGraph" -> dense;
-            default -> throw new IllegalArgumentException("Unknown graph type");
-        };
+    @Test
+    final void shouldNotAddEdgeWhenNodeIsNegative() {
```
Comment (HendrikHuebner) [inline w06h03/test/pgdp/ds/behavior/GraphTest.java:62]: I don't think this behaviour was defined on Artemis. Are you sure we need to check for this?

**Review by HendrikHuebner — APPROVED** [review]

File: w06h03/test/pgdp/ds/behavior/GraphTest.java (hunk 1 above)
Comment (julian-hartl) [reply w06h03/test/pgdp/ds/behavior/GraphTest.java]: I would rather keep it, just to be sure. Johannes [did not explicitly deny it](https://zulip.in.tum.de/#narrow/stream/1481-PGdP-W06H03/topic/from.20parameter.20g.C3.BCltig.3F).

Comment (nilsreichardt) [reply w06h03/test/pgdp/ds/behavior/GraphTest.java]: Ready to merge?

Comment (julian-hartl) [reply w06h03/test/pgdp/ds/behavior/GraphTest.java]: @nilsreichardt Let's wait for @HendrikHuebner's reply.

**Review by MaximilianAnzinger — APPROVED** [review]

File: w06h03/test/pgdp/ds/behavior/GraphTest.java (hunk 1 above)
Comment (HendrikHuebner) [reply w06h03/test/pgdp/ds/behavior/GraphTest.java]: You can go ahead and merge it, I'll ask on zulip and if they say we dont need to test for it we can still remove it later.

**Pull request merged**